### ISSUE TITLE
optional route parameter does not work

### ISSUE THREAD

**Issue by navtoj** [honojs/hono#974]
The route `/:optional?` is not used by Hono.

When i go to [`/test`](https://honojs.deno.dev/test), the optional route is correctly used by Hono.

However, when i go to [`/`](https://honojs.deno.dev/), the optional route is skipped.

## Reproduction
https://dash.deno.com/playground/honojs

## Raw Code
```ts
import { serve } from 'https://deno.land/std@0.179.0/http/server.ts';
import { logger } from 'https://deno.land/x/hono@v3.0.4/middleware.ts';
import { Hono } from 'https://deno.land/x/hono@v3.0.4/mod.ts';

// json helper
const formatJSON = (input: string) => JSON.stringify(input, null, 4);

// create a hono instance
const app = new Hono();

// middleware
app.use(logger());

// optional route
app.get('/:optional?', c => c.text(formatJSON({ route: '/:optional?', params: c.req.param() })));

// fallback route
app.get('*', c => c.text(formatJSON({ route: '*', params: c.req.param() })));

// list routes
app.showRoutes();

// start server
await serve(app.fetch);
```

**Comment by yusukebe** [comment]
Hi @navtoj !

I've released v3.0.5, including the change to resolve this issue. Try it.

**Comment by navtoj** [comment]
> I've released v3.0.5, including the change to resolve this issue. Try it.

Wow, that was fast. Thanks for the patch, the optional parameter works now.
